web/packages/teleport/src/services/apps/makeApps.ts

```diff
@@ -43,6 +43,17 @@ export default function makeApp(json: any): App {
 
   const isTcp = uri && uri.startsWith('tcp://');
   const isCloud = uri && uri.startsWith('cloud://');
+  const isGrafana =
+    name.toLocaleLowerCase().includes('grafana') ||
+    labels.some(l => `${l.name}:${l.value}` === 'icon:grafana');
+
+  const isSlack =
+    name.toLocaleLowerCase().includes('slack') ||
+    labels.some(l => `${l.name}:${l.value}` === 'icon:slack');
+
+  const isJenkins =
+    name.toLocaleLowerCase().includes('jenkins') ||
+    labels.some(l => `${l.name}:${l.value}` === 'icon:jenkins');
```

Comment:
we do something kinda similar: https://github.com/gravitational/teleport/blob/27ab9514a63e217989916b3338a189e2fa85ee1c/web/packages/shared/components/ButtonSso/ButtonSso.tsx#L72

make a function `guessAppType`?

Reply:
Good idea. i took it a step further and just `guessAppIcon` instead so we aren't guessing the type and then going through and getting an icon from the type.